web3.js/src/transaction.ts

```diff
@@ -405,6 +406,13 @@ export class Transaction {
     return this._compile().serialize();
   }
 
+  /**
+   * Get the estimated fee associated with a transaction
+   */
+  async getEstimatedFee(connection: Connection): Promise<number> {
```

Comment:
I'd would have been nice to retain the JSON RPC method name here.  It'll be an additional support burden every time we need to explain why in web3 it's `getEstimatedFee` but in JSON RPC it's `getFeeForMessage`.

If we don't like the JSON RPC method name, it's better to change that (with a deprecation of the existing method name) rather than inventing new terminology in web3.js

Reply:
It would definitely be nice to use the same terminology as core validator software and the RPC methods it exposes. `getEstimatedFee` is not a thing that exists outside of web3.js and is confusing

Reply:
Understood. I could rename it, and then you would have `getFeeForMessage` on both `Connection` and `Transaction`.

Reply:
heh, seems fine to me :)